src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs

```diff
@@ -0,0 +1,288 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedReadOnlyListTypeSymbol : NamedTypeSymbol
+    {
+        private readonly ModuleSymbol _containingModule;
+        private readonly ImmutableArray<NamedTypeSymbol> _interfaces;
+        private readonly ImmutableArray<Symbol> _members;
+
+        internal SynthesizedReadOnlyListTypeSymbol(SourceModuleSymbol containingModule, string name)
+        {
+            var compilation = containingModule.DeclaringCompilation;
+
+            _containingModule = containingModule;
+            Name = name;
+            var typeParameter = new SynthesizedReadOnlyListTypeParameterSymbol(this);
+            TypeParameters = ImmutableArray.Create<TypeParameterSymbol>(typeParameter);
+            var typeArgs = GetTypeParametersAsTypeArguments();
+            var arrayType = compilation.CreateArrayTypeSymbol(elementType: typeParameter);
+
+            // PROTOTYPE: Test missing interfaces.
+            var iEnumerable = compilation.GetSpecialType(SpecialType.System_Collections_IEnumerable);
+            var iEnumerableT = compilation.GetSpecialType(SpecialType.System_Collections_Generic_IEnumerable_T).Construct(typeArgs);
+            var iReadOnlyCollectionT = compilation.GetSpecialType(SpecialType.System_Collections_Generic_IReadOnlyCollection_T).Construct(typeArgs);
+            var iReadOnlyListT = compilation.GetSpecialType(SpecialType.System_Collections_Generic_IReadOnlyList_T).Construct(typeArgs);
+            var iCollectionT = compilation.GetSpecialType(SpecialType.System_Collections_Generic_ICollection_T).Construct(typeArgs);
+            var iListT = compilation.GetSpecialType(SpecialType.System_Collections_Generic_IList_T).Construct(typeArgs);
+
+            _interfaces = ImmutableArray.Create(
+                iEnumerable,
+                iEnumerableT,
+                iReadOnlyCollectionT,
+                iReadOnlyListT,
+                iCollectionT,
+                iListT);
+
+            // IEnumerable.GetEnumerator()
```

Comment:
plz place at the end of the constructor.  when interspersed with actual constructor logic, this is harder to understand.